src/wms/src/main/java/org/geoserver/wms/MapLayerInfo.java

```diff
@@ -240,12 +258,16 @@ public boolean isCachingEnabled() {
             return false;
         }
         ResourceInfo resource = layerInfo.getResource();
-        MetadataMap metadata = resource.getMetadata();
-        Boolean cachingEnabled = null;
-        if (metadata != null) {
-            cachingEnabled = metadata.get(ResourceInfo.CACHING_ENABLED, Boolean.class);
-        }
-        return cachingEnabled == null ? false : cachingEnabled.booleanValue();
+
+        cachingEnabled = cachingEnabled || this.isCachingEnabled(resource.getMetadata());
```

Comment:
If there is a group metadata, and it's behaving as an override, then only the group metadata is to be considered.
Otherwise, only the layer metadata should be considerd. Or-ing them would seem like they can both be considered, but it's not really the case, right?

Reply:
I think in this case it is correct.
Caching is enabled in case the layerGroup caching is enabled or the layer info caching is enabled.
Could I leave it as it is?

Reply:
Yes it is technically correct, that's why I used confusing instead of incorrect. Put yourself in the context of someone jumping in on this code without much context, what can they think it does? I would like to see code that clearly reflects the documented behavior. Believe it's easier to understand if once checked if the group header is set, then that's used, and we exit the method.